lib/gate.h

```diff
@@ -37,56 +65,95 @@ struct Gate {
 
   GateKind kind;
   unsigned time;
-  unsigned num_qubits;
   std::vector<unsigned> qubits;
+  std::vector<unsigned> controlled_by;
+  uint64_t cmask;
```

Comment:
Does this mask specify the required state of the control qubits to apply the gate? i.e. for `controlled_by = {0, 1, 2}; cmask = 0b101`, would the gate trigger for state `|101)`?

Reply:
Yes, exactly.